quic/s2n-quic-transport/src/connection/connection_container.rs

```diff
@@ -367,7 +367,11 @@ struct InterestLists<C: connection::Trait, L: connection::Lock<C>> {
     waiting_for_connection_id: LinkedList<WaitingForConnectionIdAdapter<C, L>>,
     /// Connections which are waiting for a timeout to occur
     waiting_for_timeout: RBTree<WaitingForTimeoutAdapter<C, L>>,
-    waiting_for_open: BTreeMap<InternalConnectionId, ConnectionSender>,
+    /// Connections which are waiting for a handshake to complete.
+    ///
+    /// The senders are a vector to allow multiple tasks to register interest in the same
+    /// connection being opened.
+    waiting_for_open: BTreeMap<InternalConnectionId, Vec<ConnectionSender>>,
```

Comment:
One more thing: I think it's probably worth it to use `smallvec` so we don't do an allocation for regular use. It should already be in the crate deps.